enhancements/etcd/storage-migration-for-etcd-encryption.md

```diff
@@ -0,0 +1,182 @@
+
+The first known usage is for etcd encryption at rest: `secrets` (`kube-apiserver`) and `oauthaccesstokens` (`openshift-apiserver`).
+
+## Motivation
+
+The immediate need for storage migration is to support the enablement of etcd data storage encryption. Encrypting etcd data storage involves rotating encryption keys periodically, which requires that stored data be migrated after choosing a new write key and before removing an old read key so that api servers can discard old encryption keys periodically and not lose the ability to decode the pre-existing data.
+
+We will use the upstream `kube-storage-version-migrator` (the controller-based successor to `oc adm migrate`). While upstream usage of the `kube-storage-version-migrator` is focused on migrating to new storage versions, we should be able to explicitly trigger the `kube-storage-version-migrator` when we need to migrate the encryption key used for storage, even when the storage version itself has not changed.
+
+## Goals
+
+* Make the `StorageVersionMigration` API and its corresponding controller available.
+* Invoke the `StorageVersionMigration` API from operators requesting that storage be migrated to use new etcd storage encrpyion keys.
+
+## Non-Goals
+
+* Will not provide a higher level API to the `StorageVersionMigration` api controlled by the `kube-storage-version-migrator`.
+* Will not implement any general control loops to automatically detect and then initiate a storage migration when needed.
+
+## Proposal
+
+* Introduce `openshift/cluster-openshift-kube-storage-version-migrator-operator` repository.
+  * Not required for bootstrapping or re-bootstrapping.
+  * No need for static pods.
+  * Will run on cluster.
+* Implement `openshift-kube-storage-version-migrator-operator`. 
+* Build `kube-storage-version-migrator` image.
+  * Fork `kubernetes-sigs/kube-storage-version-migrator` to `openshift`
+  * Enable CI on `openshift/kube-storage-version-migrator` repo.
+* Utilize `StorageVersionMigration` API when etcd storage encryption keys are updated.
+
+## User Stories [optional]
+
+* kube-apiserver-operator can request a resource migration and confirm when the migration has completed. 
+
+* openshift-apiserver-operator can request a resource migration and confirm when the migration has completed. 
+
+## Implementation Plan
+
+1. Implement `openshift-kube-storage-version-migrator-operator`.
+2. Have CVO manage `openshift-kube-storage-version-migrator-operator`.
+2. Add control loops to operators as needed.
+
+## Implementation Details/Notes/Constraints
+
+* Create a `openshift-kube-storage-version-migrator-operator`, managed by the CVO, to manage the resources needed to run the `kube-storage-version-migrator`.
+* Initiate a storage migration of resources, even if the storage version has not changed.
```

Comment:
is this a new feature/capability of the controller?

Reply:
No. I just wanted to be clear, since the `StorageVersionMigration` focuses on the actual storage version and we don't really care about the storage version.

Reply:
My question was: do we need upstream changes for that?